eulerpi/core/plotting.py

```diff
@@ -0,0 +1,393 @@
+"""Basic plotting of eulerpi sampling results.
+
+This module provides a basic plotting functionality to visualise sampling results for eulerpi.
+Uses burn_in and thinning accordinng to the simulation settings.
+"""
+
+from typing import Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib import axes
+
+from eulerpi.core.kde import calc_kernel_width, eval_kde_gauss
+from eulerpi.core.model import Model
+from eulerpi.core.result_manager import ResultManager
+
+# general plotting function for joint runs
+
+
+def sample_violin_plot(
+    model: Model,
+    run_name: str = "default_run",
+    what_to_plot: str = "param",
+    artificial_bool: bool = False,
+    credibility_level: float = 0.95,
+    num_vertical_grid_points: int = 100,
+    axis_labels: Optional[list[str]] = None,
+) -> axes:

```

Comment:
Following our discussion, I would propose to use the bool variables related to `artificial_bool` only internally and offer an interface similar to the inference function to the user.
This would mean to allow the user to pass something like true_samples: Optional[Union[str, os.PathLike, np.ndarray]] = None. Then we can use the loading snippet from the inference function and define reference_available from the value of what_to_plot and whether true_samples is None.

Reply:
I like the idea and will work on it.